src/main.js

```diff
@@ -322,17 +322,29 @@ class MiniGraphCard extends LitElement {
     `;
   }
 
+  renderLoader() {
+    return html`
+        <div style="display: flex;">
+             <ha-circular-progress indeterminate style="margin: auto;"></ha-circular-progress>
+        </div>
+    `;
+  }
+
   renderGraph() {
+    const ready = this.entity[0] && this.Graph[0]._history !== undefined;
+
     return this.config.show.graph ? html`
       <div class="graph">
-        <div class="graph__container">
-          ${this.renderLabels()}
-          ${this.renderLabelsSecondary()}
-          <div class="graph__container__svg">
-            ${this.renderSvg()}
-          </div>
-        </div>
-        ${this.renderLegend()}
+        ${ready ? html`
+            <div class="graph__container">
+              ${this.renderLabels()}
+              ${this.renderLabelsSecondary()}
+              <div class="graph__container__svg">
+                ${ready ? this.renderSvg() : this.renderLoader()}
```

Comment:
I think, we can safely always render the SVG here, now. Can't we?

Reply:
> always render the SVG here,

But this is the only place we are checking for `ready`, no?

Reply:
It's a leftover, I've removed it

Reply:
We have a new check to omit the entire graph__container. So, this one here is now duplicated.

Reply:
I think, you accidentally also removed the `${}` part...

Reply:
I don't know what you're talking about .. 😅
(fixed)

Reply:
lots of smileys here
![image](https://github.com/user-attachments/assets/f8d0f6a3-9784-48fa-a527-8357030decdf)